libsmtutil/Z3Interface.cpp

```diff
@@ -371,7 +372,10 @@ Expression Z3Interface::fromZ3Expr(z3::expr const& _expr)
 		return Expression("dt_accessor_" + _expr.decl().name().str(), arguments, sort);
 	else if (kind == Z3_OP_DT_IS)
 		return Expression("dt_is", {arguments.at(0)}, sort);
-	else if (kind == Z3_OP_UNINTERPRETED)
+	else if (
+		kind == Z3_OP_UNINTERPRETED ||
+		kind == Z3_OP_RECURSIVE
+	)
```

Comment:
Is it possible to add some kind of test to verify that it's handled correctly?

What is the result of just feeding the expression string directly into `Expression`? Will it just somehow go through without issues or can something in our code fail when it's encountered?

Reply:
With (2) I was actually asking what happens if we do `Expression(_expr.decl().name().str(), ...)` and it turns out the string was something invalid. Will it crash at some point or just go unnoticed. I'm asking because I have no idea if this is the right way to handle this and if it will fail with something invalid, it would at least give me more confidence that the fix is not wrong :)

Reply:
> I have a tendency to keep it as is because this is the approach we used early on to detect what kind of AST nodes we should handle, but fine with either if you think the second option is safer.

We could actually have the compiler detect any unhandled values at compile time if you replaced the `if` with a `switch` (could be a separate PR though). It would not catch the issue at runtime though so I guess we still need the assert even then.

Reply:
Ah, ok. We have so many listed that I thought it's pretty much all of them. So those we don't have actually never happen with our queries?

Reply:
> Ah no, the string should be fine. From what I read in their docs/code both are functions, but the RECURSIVE one has a `recursive` flag.

OK then. Would be best to have it tested but if we can't really trigger it easily then I guess there's not much we can do.

Reply:
> Ah, ok. We have so many listed that I thought it's pretty much all of them. So those we don't have actually never happen with our queries?

Exactly.

Reply:
> OK then. Would be best to have it tested but if we can't really trigger it easily then I guess there's not much we can do.

Yea I wouldn't really know how to trigger it on purpose. I can try to do that with more time next week.